tools/istio-iptables/pkg/dependencies/implementation.go

```diff
@@ -71,11 +73,28 @@ type RealDependencies struct {
 }
 
 func (r *RealDependencies) execute(cmd string, ignoreErrors bool, args ...string) error {
-	if r.CNIMode {
-		originalCmd := cmd
-		cmd = constants.NSENTER
-		args = append([]string{fmt.Sprintf("--net=%v", r.NetworkNamespace), "--", originalCmd}, args...)
+	// Lock the OS Thread so we don't accidentally switch namespaces
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	ns, err := netns.GetFromPath(r.NetworkNamespace)
+	if err != nil {
+		return err
+	}
+	defer ns.Close()
+
+	origns, err := netns.Get()
+	if err != nil {
+		return err
 	}
+	defer origns.Close()
+
+	err = netns.Set(ns)
```

Comment:
should we use `Do` to avoid the complex sequence here: https://github.com/containernetworking/plugins/blob/main/pkg/ns/README.md#do-the-recommended-thing? It seems recommended

Reply:
Yes, makes perfect sense. will do